packages/contractkit/src/utils/web3-utils.ts

```diff
@@ -71,3 +71,68 @@ class WrappingSubprovider extends Subprovider {
     })
   }
 }
+
+// tslint:disable-next-line:max-classes-per-file
+export class CeloProvider implements Provider {
+  private static createNewProviderWithLocalAccount(existingProvider: Provider): Web3ProviderEngine {
+    // Create a Web3 Provider Engine
+    const providerEngine = new Web3ProviderEngine()
+    // Use the existing provider to route all other requests
+    const wrappingSubprovider = new WrappingSubprovider(existingProvider)
+    debug('Setting up providers...')
+    providerEngine.addProvider(wrappingSubprovider)
+    return providerEngine
+  }
+
+  private web3ProviderEngine: Web3ProviderEngine
+  private signingProviders: CeloPrivateKeyWalletProvider[]
+  private signingAddresses: string[]
+
+  constructor(readonly existingProvider: Provider) {
+    this.web3ProviderEngine = CeloProvider.createNewProviderWithLocalAccount(existingProvider)
+    this.signingProviders = []
+    this.signingAddresses = []
+  }
+
+  addAccount(privateKey: string): CeloProvider {
+    if (this.signingAddresses.includes(privateKey)) {
+      debug('Account is already added')
+      return this
+    }
+
+    const localAccountProvider = new CeloPrivateKeyWalletProvider(privateKey)
```

Comment:
imo, a better strategy to is have just ONE `CeloPrivateKeyWalletProvider` that handles many accounts. There a couple of things/state that they share (liked cached values for chainId and others).

This also avoids modifying the providerEngine providers list, mantain a list of providers, etc...

In fact, we can even store the accounts in the CeloProvider and pass a `getAccounts()` fn to the CeloPrivateKeyWalletProvider